### PR TITLE
Issue/related posts rename

### PR DESCRIPTION
**Pull request by nbradbury**
TL;DR: This PR doesn't change any logic, it just renames a lot of stuff.

Long version:

I started adding a "recommended posts" feature to the reader stream view, and as I got into it I realized I could reuse a lot of the logic/ui I'd already written for "related posts" (which appear in the full post view). 

Since the code was written and named with related posts in mind, it made sense to rename it for more generic re-use. Rather than do that as part of the "recommended posts" feature (and make that PR huge), I figured I'd do it as a separate PR.

To test: go to the full post ("detail") view and scroll down to the related posts section and make sure it still works. This is the section with "More in [site name]" and "More on WordPress.com."

### CODE REVIEW

**Comment by daniloercoli** [comment]
🚢
